### PR TITLE
 TW-850 In Chat, user can click to avatar of sender to see user info

### PR DESCRIPTION
**Pull request by drminh2807**
# Issue
- #850 

# Demo

https://github.com/linagora/twake-on-matrix/assets/12546908/61b5412d-4f7d-4d15-b983-dc10c3db5bdd

### CODE REVIEW

File: lib/pages/chat/events/html_message.dart [hunk 1]
```diff
@@ -77,7 +77,8 @@ class HtmlMessage extends StatelessWidget {
       shrinkToFit: true,
       maxLines: maxLines,
       onLinkTap: (url) => UrlLauncher(context, url.toString()).launchUrl(),
-      onPillTap: (url) => UrlLauncher(context, url).launchUrl(),
+      onPillTap: chatController?.onPillTap ??
```
Comment (sherlockvn) [inline lib/pages/chat/events/html_message.dart:80]: onTagTap

Comment (drminh2807) [reply lib/pages/chat/events/html_message.dart]: Fixed in [336b77c](https://github.com/linagora/twake-on-matrix/pull/1007/commits/336b77c1427b32d19e2b3cf6f5cdf98d20b0ce62)

File: lib/pages/chat/chat.dart [hunk 2]
```diff
@@ -1572,8 +1573,26 @@ class ChatController extends State<Chat>
...
+  void onAvatarTap(Event event) {
+    openProfileInformation(event.senderId);
+  }
+
+  void openProfileInformation(String userId) {
+    widget.onChangeRightColumnType
+        ?.call(ProfileInfoRightColumnType(userId: userId));
+  }
+
+  void onPillTap(String url) {
```
[the remark this answers is not in the document]
Comment (drminh2807) [reply lib/pages/chat/chat.dart]: Fixed in [336b77c](https://github.com/linagora/twake-on-matrix/pull/1007/commits/336b77c1427b32d19e2b3cf6f5cdf98d20b0ce62)

**Review by sherlockvn — APPROVED** [review]

**Comment by hoangdat** [comment]
we should not apply it, when click on avatar in group chat -> open direct chat (draft chat)

**Comment by drminh2807** [comment]
> we should not apply it, when click on avatar in group chat -> open direct chat (draft chat)

Refered Telegram
- Web: It's true
- Mobile: It will open profile info

File: lib/pages/profile_info/profile_info.dart [hunk 3]
```diff
@@ -27,8 +29,9 @@ class ProfileInfoController extends State<ProfileInfo> {
       ? Matrix.of(context).client.getRoomById(widget.roomId!)
       : null;
 
-  User? get user =>
-      room?.unsafeGetUserFromMemoryOrFallback(room?.directChatMatrixID ?? '');
+  User? get user => room?.unsafeGetUserFromMemoryOrFallback(
+        room?.directChatMatrixID ?? widget.userId ?? '',
```
Comment (imGok) [inline lib/pages/profile_info/profile_info.dart:33]: Can you explain why it's not `widget.userId` first ?

File: lib/widgets/avatar/avatar.dart [hunk 4]
```diff
@@ -35,22 +35,26 @@ class Avatar extends StatelessWidget {
...
-          boxShadows: boxShadows,
-          textStyle: TextStyle(
-            fontSize: fontSize,
-            color: textColor ?? AvatarStyle.defaultTextColor(_havePicture),
-            fontFamily: AvatarStyle.fontFamily,
-            fontWeight: AvatarStyle.fontWeight,
+      child: InkWell(
+        onTap: onTap,
+        customBorder: const CircleBorder(),
+        child: MxcImage(
```
Comment (imGok) [inline lib/widgets/avatar/avatar.dart:41]: MxcImage has already an inkwell why not adding an `onTap` parameter that overrides the native one ? Also you can use `rounded: true` for your borders.

IMO it's not good to add clickable area above other clickable areas

File: lib/pages/profile_info/profile_info.dart (hunk 3 above)
Comment (drminh2807) [reply lib/pages/profile_info/profile_info.dart:33]: no worries, room or userId cannot both not null, so the priority is same

File: lib/widgets/avatar/avatar.dart (hunk 4 above)
Comment (drminh2807) [reply lib/widgets/avatar/avatar.dart:41]: rounded is just borderRadius 12, not a circle
MxcImage also dont have onTap

Comment (imGok) [reply lib/widgets/avatar/avatar.dart:41]: we have `onTapPreview` if "isPreview" is true but what I mean is we could refactor mxcImage so we can give it an optional onTap if needed and get rid of onTapPreview. Anyway it was just my idea, close the comment if you dont like it ! :)

Comment (drminh2807) [reply lib/widgets/avatar/avatar.dart:41]: MxcImage should be refactor in onather PR
It impact many screens

**Review by nqhhdev — APPROVED** [review]

**Pull request closed without merging**